File: java/src/main/java/ai/buble/sdk/media/MediaModel.java
package ai.buble.sdk.media;

import java.util.List;

/**
 * API-ready media model returned by media model discovery.
 */
public class MediaModel {
    private String model;
    private String name;
    private String mediaType;
    private List<MediaModelOperation> operations;

    public String getModel() { return model; }
    public void setModel(String model) { this.model = model; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getMediaType() { return mediaType; }
    public void setMediaType(String mediaType) { this.mediaType = mediaType; }
    public List<MediaModelOperation> getOperations() { return operations; }
    public void setOperations(List<MediaModelOperation> operations) { this.operations = operations; }
}

File: dotnet/src/Buble.Sdk/Media/MediaModel.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Buble.Sdk.Media;

public sealed class MediaModel
{
    [JsonPropertyName("model")]
    public string? Model { get; set; }

    [JsonPropertyName("name")]
    public string? Name { get; set; }

    [JsonPropertyName("media_type")]
    public string? MediaType { get; set; }

    [JsonPropertyName("operations")]
    public IReadOnlyList<MediaModelOperation>? Operations { get; set; }

    [JsonExtensionData]
    public Dictionary<string, JsonElement>? AdditionalProperties { get; set; }
}
